sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -389,94 +392,83 @@ class tf32;
 
-template <typename Group, size_t Rows, size_t Cols,
-          access::address_space Space, access::decorated IsDecorated>
-void joint_matrix_load(Group g,
-    joint_matrix<Group, precision::tf32, use::accumulator, Rows, Cols,
-    layout::dynamic> &res,
-    multi_ptr<float, Space, IsDecorated> src, size_t stride, layout Layout);
+joint_matrix_load(sg, tA, pBuf + Offset, Stride);
+```
 
-// Only available when Layout != layout::dynamic
-template <typename Group, size_t Rows, size_t Cols,
-          use Use, layout Layout,
-          access::address_space Space, access::decorated IsDecorated>
-void joint_matrix_load(Group g,
-    joint_matrix<Group, precision::tf32, Use, Rows, Cols, Layout> &res,
-    multi_ptr<const float, Space, IsDecorated> src, size_t stride);
+==== store
+The template parameter `T2` must be the storage format type that
+corresponds to the abstract type `T1`. For example:
 
-// Only available when Layout != layout::dynamic
-template <typename Group, size_t Rows, size_t Cols,
-          use Use, layout Layout,
-          access::address_space Space, access::decorated IsDecorated>
-void joint_matrix_load(Group g,
-    joint_matrix<Group, precision::tf32, Use, Rows, Cols, Layout> &res,
-    multi_ptr<float, Space, IsDecorated> src, size_t stride);
+```c++
+joint_matrix<sub_group, precision::tf32, use::accumulator, tM, tK> tC;
 
-} // namespace sycl::ext::oneapi::experimental::matrix
+float *buf = malloc_shared<float>(M*K, q);
+auto pBuf = address_space_cast<sycl::access::address_space::global_space,
+                               sycl::access::decorated::no>(buf);
+
+joint_matrix_store(sg, tA, pBuf + Offset, Stride, layout::row_major);
 ```
 
-==== TF32 store
-This overload of joint_matrix_store stores float values from a TF32
-matrix.
+==== fill
+The template parameter `Tv` must be implicitly convertible to the
+storage format type that corresponds to the abstract type `T`. For example:
 
 ```c++
-namespace sycl::ext::oneapi::experimental::matrix {
-
-template <typename Group, size_t Rows, size_t Cols,
-          access::address_space Space, access::decorated IsDecorated>
-void joint_matrix_store(Group g,
-   const joint_matrix<Group, precision::tf32, use::accumulator, Rows,
-                      Cols,  layout::dynamic> &res,
-   multi_ptr<float, Space, IsDecorated> dest, size_t stride, layout Layout);
-
-} // namespace sycl::ext::oneapi::experimental::matrix
+joint_matrix<sub_group, precision::tf32, use::a, tM, tK, layout::row_major> tA;
+float v = 42.0;
+joint_matrix_fill(sg, tA, v);
 ```
 
-==== TF32 fill
-When `joint_matrix_fill` is called for a TF32 matrix, the type `Tv`
-(the type of the fill value) must be implicitly convertible to
-`float`. It is unspecified whether the implementation writes all 32
-bits of the value into the joint matrix or if it only writes the
-relevant 19 bits.
+==== copy
+The template parameters `T1` and `T2` must use the abstract type.
```

Comment:
After looking at your example, I think this wording could be improved.  Note that the example shows a case when `T2` is *not* an abstract type.  Instead, I'd say it this way:

> There is no special constraint for the `joint_matrix_copy` function.  The template parameters `T1` and `T2` correspond to the element types of the `src` and `dest` matrices.